### PR TITLE
chore: allow dependency changes from non-maintainers with label

### PR DESCRIPTION
**Pull request by samuelmaddock**
Sometimes dependency changes might be necessary from non-maintainers. See https://github.com/electron/electron/pull/42001#issuecomment-2212383114

Adds a conditional which skips this check when `allow-dependency-change` is an added label.

#### Release Notes

Notes: none

### CODE REVIEW

File: .github/workflows/auto-close-pull-request.yml
Comment (dsanders11) [inline .github/workflows/auto-close-pull-request.yml:19]:
```suggestion
        !contains(github.event.pull_request.labels.*.name, 'allow-dependency-change')
```

https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter

Comment (samuelmaddock) [reply .github/workflows/auto-close-pull-request.yml:19]: Thank you, I was unsure about how to test this 🙇

**Review by MarshallOfSound — CHANGES_REQUESTED** [review]
This policy should not be **overridden** rather we should find a way to ensure this can be done safely under the existing policy.

For instance, ensuring that the diff in `yarn.lock` is entirely committed by a maintainer and allowing maintainers to push to the PR.

**Comment by reitowo** [comment]
I think that will cause a lot of burden to contributor considering the average reply time of maintainers.

- Do I need to seperate only `yarn.lock` and `package.json` changes to a seperate diff and wait for a maintainer to manually commit this to my own branch? 
- If I made further changes it seems I have to manually rebase or re-generate the diff every time, and wait for a maintainer to apply that. Otherwise I have to be very careful and let the depencency change as the very first commit.
- How do you ensure the commit related to dependency changes are commited by maintainer?
- It also prevent contributors from force-pushing commits. 

Adding a override label seems good to me as the label can only added by maintainer, and maintainers have final decision about commited changes. I don't see a reason for making things so complicated as described above.

**Comment by samuelmaddock** [comment]
closing in favor of https://github.com/electron/electron/pull/42855

**Pull request closed without merging**
